internal/k8s/statusupdater.go

```diff
@@ -0,0 +1,186 @@
+// Copyright © 2020 VMware
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8s
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/projectcontour/contour/internal/assert"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// StatusUpdate contains an all the information needed to change an object's status to perform a specific update.
+// Send down a channel to the goroutine that actually writes the changes back.
+type StatusUpdate struct {
+	FullName FullName
```

Comment:
Suggest just `Name` for this field?

Reply:
I'm leaving it as FullName, so that when you're retrieving the name you get `x.FullName.Name` and `x.FullName.Namespace`, rather than `x.Name.Name` and `x.Name.Namespace`.